src/pymor/parameters/functionals.py

```diff
@@ -234,3 +235,90 @@ def evaluate(self, mu=None):
 
     def d_mu(self, component, index=()):
         return self.with_(constant_value=0, name=self.name + '_d_mu')
+
+
+class MinThetaParameterFunctional(ParameterFunctionalInterface):
+    """|ParameterFunctional| implementing the min-theta approach from [Haa17]_ (Proposition 2.35).
+
+
+    Parameters
+    ----------
+    thetas
+        List or tuple of |ParameterFunctional|s
+
+    mu_bar
+        see [Haa17]_ (Proposition 2.35)
+
+    alpha_mu_bar
+        see [Haa17]_ (Proposition 2.35)
+
+    name
+        Name of the functional.
+    """
+
+    def __init__(self, thetas, mu_bar, alpha_mu_bar=1., name=None):
+        assert isinstance(thetas, (list, tuple))
+        assert all([isinstance(theta, (Number, ParameterFunctionalInterface)) for theta in thetas])
+        thetas = [ConstantParameterFunctional(theta) if not isinstance(theta, ParameterFunctionalInterface) else theta
+                  for theta in thetas]
+        assert all([isinstance(f, ParameterFunctionalInterface) for f in thetas])
```

Comment:
After the line above this assertion does not make much sense. Probably you want to assert before that you only have `Numbers` and `ParameterFunctionals`.